File: level2/더맵게.py
import heapq

def solution(scoville, K):
    heap = []
    for s in scoville:
        heapq.heappush(heap,s)
    
    answer = 0
    while heap[0] < K:   
        answer += 1
        f1 = heapq.heappop(heap)
        if not heap:
            answer = -1
            break
        f2 = heapq.heappop(heap)
        
        heapq.heappush(heap,f1+f2*2)
            
    return answer

File: level2/더맵게.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> scoville, int K) {
    priority_queue<int, vector<int>, greater<int>> pq;
    for(int i = 0; i < scoville.size(); i++){
        pq.push(scoville[i]);
    }
    
    int answer = -1, tmp = 0;
    while(!pq.empty()){
        int s1, s2;
        s1 = pq.top();
        pq.pop();
        
        if(s1 >= K){
            answer = tmp;
            break;
        }
        
        if(pq.empty()){
            break;
        }
        tmp++;
        
        s2 = pq.top();
        pq.pop();
        pq.push(s1+s2*2);      
    }
    return answer;
}
